core/domain/story_fetchers.py

```diff
@@ -130,7 +144,43 @@ def get_story_summary_from_model(story_summary_model):
     )
 
 
-def get_story_by_id(story_id, strict=True, version=None):
+@overload
+def get_story_by_id(
```

Comment:
Feel free to reorder the args here if it reduces the amount of overloads.

Reply:
Even if I change the order of args the total number of overloads remains the same ( which is 4 ).